### ISSUE TITLE
On linux, Iris server hangs with Temporary error when accepting new connections: accept tcp4 

### ISSUE THREAD

**Issue by antlaw** [kataras/iris#560]
- Version : **5.0.2**
- go version 1.7.3
- Operating System: Ubuntu 14.04.5 LTS

- [x] I wrote an http server to handle simple http get/set requests. The program hangs when running on Ubuntu 14.04.5 LTS. It can handle requests without problem for a while, but hangs the incoming requests after certain time. The server outputs something like the following:
- When I run the same program on Windows 10 + iris 5.0.2, the problem will not occur.
- The testing http get requests don't even access database, therefore I suspect that the problem comes from iris.
INFO[0000] confg:                                        host= port=28080
         _____      _
        |_   _|    (_)
          | |  ____ _  ___
          | | | __|| |/ __|
         _| |_| |  | |\__ \
        |_____|_|  |_||___/ 4

Wed, 14 Dec 2016 13:04:52 GMT: Running at 0.0.0.0:28080
2016/12/14 15:29:55 Temporary error when accepting new connections: accept tcp4                  0.0.0.0:28080: accept4: too many open files
2016/12/14 15:29:56 Temporary error when accepting new connections: accept tcp4                  0.0.0.0:28080: accept4: too many open files
2016/12/14 15:29:57 Temporary error when accepting new connections: accept tcp4                  0.0.0.0:28080: accept4: too many open files
2016/12/14 15:29:58 Temporary error when accepting new connections: accept tcp4                  0.0.0.0:28080: accept4: too many open files
2016/12/14 15:29:59 Temporary error when accepting new connections: accept tcp4                  0.0.0.0:28080: accept4: too many open files
2016/12/14 15:30:00 Temporary error when accepting new connections: accept tcp4                  0.0.0.0:28080: accept4: too many open files
2016/12/14 15:30:01 Temporary error when accepting new connections: accept tcp4                  0.0.0.0:28080: accept4: too many open files
2016/12/14 15:30:02 Temporary error when accepting new connections: accept tcp4                  0.0.0.0:28080: accept4: too many open files
2016/12/14 15:30:03 Temporary error when accepting new connections: accept tcp4                  0.0.0.0:28080: accept4: too many open files
2016/12/14 15:30:04 Temporary error when accepting new connections: accept tcp4                  0.0.0.0:28080: accept4: too many open files
2016/12/14 15:30:05 Temporary error when accepting new connections: accept tcp4                  0.0.0.0:28080: accept4: too many open files
2016/12/14 15:30:06 Temporary error when accepting new connections: accept tcp4                  0.0.0.0:28080: accept4: too many open files
2016/12/14 15:30:07 Temporary error when accepting new connections: accept tcp4                  0.0.0.0:28080: accept4: too many open files

- 

--------------

**Comment by kataras** [comment]
The iris is auto-testing on multi platforms before of each release(or mini version) on:

- BSD( PcBSD)
- Linux ( Ubuntu 16.04)
- Windows 7 SP1
- Windows 10
- MacOS Siera

All on virtual box which is managed by script I wrote years ago and all tests must pass before a commit (on .go files).

Please provide us more information in order to help you, I , myself can't help you if you don't write a code snippet at least... Thanks!!

**Comment by kataras** [comment]
The iris is auto-testing on multi platforms before of each release(or mini version) on:

- BSD( PcBSD)
- Linux ( Ubuntu 16.04)
- Windows 7 SP1
- Windows 10
- MacOS Siera

All on virtual box which is managed by script I wrote years ago and all tests must pass before a commit (on .go files).

Please provide us more information in order to help you, I , myself can't help you if you don't write a code snippet at least... Although this question is not about Iris but about [fasthttp protocol library](https://github.com/valyala/fasthttp) go there and search for an answer on its issue lists I think I saw that before and its solution was simple. Thanks!!

**Comment by antlaw** [comment]
Sorry for replying late to the post.

After quite some tests to isolate the problem, we found that the problem is
not from iris, but from the incompatibility issue of database library to
the database version (we are using couchbase). The reason that we thought
it was related to iris was that the error message was from fast http that
called from iris. However, that was not correct. We noticed that removing
the database library codes removes the problem as well. Finally, we tried
upgrade the database and the problem was gone. Anyway, thx very much for
your reply.

Merry X'mas!

Best,

Anthony LAW
